microprot/scripts/pdb_search.py

```diff
@@ -388,3 +391,134 @@ def report_uncovered_subsequences(hits, query, min_subseq_len=40):
+    """ Splits a protein sequence according to HHsuits results.
+
+    The returned sub-sequences will seamlessly build the full sequence if
+    re-concatenated.
+
+    Parameters
+    ----------
+    hhsuite_fp : str
+        Filepath to HHblits/HHsearch output.
+    fullsequence_fp : str
+        Filepath to the protein sequence of the original query.
+    subsequences_fp : str
+        Filepath to which sub-sequences are written as a multiple fasta file.
+        Each sequence makes up one header and one sequence file, i.e. sequences
+        are not wrapped.
+        Default: None, i.e. no file is written.
+    min_prob: float
+        Minimal probability of a hit to be included in the resulting list.
+        Note: probabilities are in the range of 100.0 to 0.0.
+        Default: None, i.e. no filtering on probability.
+    max_pvalue: float
+        Maximal P-value of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on P-value.
+    max_evalue: float
+        Maximal E-value of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on E-value.
+    min_fragment_length: int
+        Minimal fragment length of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on fragment length.
+
+    Returns
+    -------
+    [(str, str)] where first component is a fasta header, while the second is
+    its fasta sequence.
+
+    Raises
+    ------
+    IOError
+        If the file cannot be written.
+
+    Notes
+    -----
+    A hit must satisfy ALL filtering options (min_prob, max_pvalue, max_evalue,
+    min_fragment_length) to be included in the resulting list.
+    """
+
+    # parse hits from file
+    hits = parse_pdb_match(hhsuite_fp)
+
+    # filter hits
+    if min_prob is not None:
+        hits = [hit for hit in hits if hit['Probab'] >= min_prob]
+    if max_pvalue is not None:
+        hits = [hit for hit in hits if hit['P-value'] <= max_pvalue]
+    if max_evalue is not None:
+        hits = [hit for hit in hits if hit['E-value'] <= max_evalue]
+    if min_fragment_length is not None:
+        hits = [hit for hit in hits if frag_size(hit) >= min_fragment_length]
+
+    # read the original protein file, used to run HHsearch
+    p = Protein.read(fullsequence_fp, seq_num=1)
+    queryname = p.metadata['id'] + " " + p.metadata['description']
+
+    results = {'match': [], 'non_match': []}
+    # select non overlapping positive hits
+    subseqs_pos = select_hits(hits, e_value_threshold=999999)
+
+    for hit in subseqs_pos:
+        header = "%s_%i-%i" % (queryname,
+                               hit['alignment']['Q Consensus']['start'],
+                               hit['alignment']['Q Consensus']['end'])
+        seq = hit['alignment']['Q Consensus']['sequence'].replace('-', '')
+        results['match'].append((header, seq,
+                                hit['alignment']['Q Consensus']['start']))
```

Comment:
actually, I'd be more comfortable if all `'Q Consensus'` would become `'Q SequenceName'`.